File: API/PetCenterModels/DataTransferObjects/ImageDTO.cs
using Microsoft.VisualBasic;
using PetCenterModels.DBTables;
using PetCenterModels.DataTransferObjects;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;
using PetCenterModels.ModelUtils;

namespace PetCenterModels.DataTransferObjects
{
    public class ImageDTO : IBLOBReferencingDTO<Image,ImageDTO,ImageMetadata>
    {
               
        public Guid? Id {get; set;} = null;

        public byte[] CurrentVersion { get; set; } = Array.Empty<byte>();

        [Required]
        public Guid AlbumInsertId { get; set; } = Guid.Empty;
        [Required]
        public short Width { get; set; } = 0;
        [Required]
        public short Height { get; set; } = 0;

        public string? Token {get; set;} = null;

        public string Hash {get; set;} = string.Empty;   

        public bool CanWrite {get; set;} = false;
  

        public List<NoteSubDTO>? Notes {get; set;} = null;

        public static ImageDTO? FromEntity(Image? img)
        {
            if(img==null){return null;}
            return new ImageDTO
            {
                Id = img.Id,
                CurrentVersion=img.CurrentVersion,
                AlbumInsertId=img.AlbumId,
                Width = img.Width,
                Height = img.Height,
                Hash = img.BLOBId
            };
        }

        public static ImageDTO? FromEntity(Image? img, string token)
        {
            ImageDTO? output = ImageDTO.FromEntity(img);

            if (output != null)
            {
                output.Token=token;
            }

            return output;
        }

    }
}

File: Frontend/pet_center_app/lib/models/data_transfer/image_dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pet_center_app/models/data_transfer/note_sub_dto.dart';

part 'image_dto.g.dart';

@JsonSerializable(explicitToJson: true)
class ImageDTO {
  String? id;
  String currentVersion;
  String albumInsertId;
  int width;
  int height;
  String? token;
  String hash;
  bool canWrite;
  List<NoteSubDTO>? notes;
  ImageDTO({
    this.id,
    this.currentVersion = '',
    this.albumInsertId = '',
    this.width = 64,
    this.height = 64,
    this.token,
    this.hash = '',
    this.notes,
    this.canWrite = false,
  });

  factory ImageDTO.fromJson(Map<String, dynamic> json) =>
      _$ImageDTOFromJson(json);

  Map<String, dynamic> toJson() => _$ImageDTOToJson(this);
}
